pkg/network/tracer/tracer_usm_linux_test.go

```diff
@@ -2278,14 +2282,22 @@ func testProtocolClassificationLinux(t *testing.T, tr *Tracer, clientHost, targe
 	}
 }
 
+// tryGoTLSAttachPID tries to attach the Go-TLS monitoring to the given PID.
+func tryGoTLSAttachPID(t *testing.T, pid int) error {
+	t.Helper()
+	if utils.IsProgramTraced("go-tls", pid) {
+		return nil
+	}
+	return usm.GoTLSAttachPID(uint32(pid))
+}
+
 // goTLSAttachPID attaches the Go-TLS monitoring to the given PID.
 // Wraps the call to the Go-TLS attach function and waits for the program to be traced.
 func goTLSAttachPID(t *testing.T, pid int) {
 	t.Helper()
-	if utils.IsProgramTraced("go-tls", pid) {
-		return
-	}
-	require.NoError(t, usm.GoTLSAttachPID(uint32(pid)))
+	require.Eventually(t, func() bool {
+		return tryGoTLSAttachPID(t, pid) == nil
```

Comment:
Maybe `EventuallyWithT` and `assert.NoError` would be better so that the error code is printed if the Eventually loop fails?